soc/xtensa/esp32/CMakeLists.txt

```diff
@@ -3,4 +3,148 @@
+
+  board_finalize_runner_args(esp32 "--esp-app-address=0x10000")
+endif()
+
+if(CONFIG_BOOTLOADER_MCUBOOT)
+  include(ExternalProject)
+  set(mcuboot_dir ${ZEPHYR_BASE}/../bootloader/mcuboot)
+  set(mcuboot_esp_dir ${mcuboot_dir}/boot/espressif)
+  set(mcuboot_build_dir ${CMAKE_BINARY_DIR}/mcuboot)
+
+  file(COPY ${CMAKE_CURRENT_LIST_DIR}/mcuboot.defconfig DESTINATION ${mcuboot_build_dir})
+
+  dt_nodelabel(dts_partition_path NODELABEL "boot_partition")
+  dt_reg_size(mcub_size PATH ${dts_partition_path})
+  file(APPEND ${mcuboot_build_dir}/mcuboot.defconfig
+    "CONFIG_ESP_BOOTLOADER_SIZE=${mcub_size}\n")
+
+  dt_nodelabel(dts_partition_path NODELABEL "slot0_partition")
+  dt_reg_addr(img_0_off PATH ${dts_partition_path})
+  dt_reg_size(img_size PATH ${dts_partition_path})
+  file(APPEND ${mcuboot_build_dir}/mcuboot.defconfig
+    "CONFIG_ESP_APPLICATION_PRIMARY_START_ADDRESS=${img_0_off}\n"
+    "CONFIG_ESP_APPLICATION_SIZE=${img_size}\n")
+
+  dt_nodelabel(dts_partition_path NODELABEL "slot1_partition")
+  dt_reg_addr(img_1_off PATH ${dts_partition_path})
+  file(APPEND ${mcuboot_build_dir}/mcuboot.defconfig
+    "CONFIG_ESP_APPLICATION_SECONDARY_START_ADDRESS=${img_1_off}\n"
+  )
+
+  dt_nodelabel(dts_partition_path NODELABEL "scratch_partition")
+  dt_reg_addr(img_scr_off PATH ${dts_partition_path})
+  dt_reg_size(img_scr_size PATH ${dts_partition_path})
+  file(APPEND ${mcuboot_build_dir}/mcuboot.defconfig
+    "CONFIG_ESP_SCRATCH_OFFSET=${img_scr_off}\n"
+    "CONFIG_ESP_SCRATCH_SIZE=${img_scr_size}\n"
+  )
+
+  if(CONFIG_MCUBOOT_GENERATE_CONFIRMED_IMAGE)
+    set(app_bin ${CONFIG_KERNEL_BIN_NAME}.signed.confirmed.bin)
+  else()
+    set(app_bin ${CONFIG_KERNEL_BIN_NAME}.signed.bin)
+  endif()
+
+  board_finalize_runner_args(esp32
+    "--esp-flash-bootloader=${mcuboot_build_dir}/mcuboot_esp32.bin")
+  board_finalize_runner_args(esp32 "--esp-boot-address=0x1000")
+  board_finalize_runner_args(esp32 "--esp-app-address=${img_0_off}")
+  board_finalize_runner_args(esp32
+    "--esp-flash-app=${CMAKE_BINARY_DIR}/zephyr/${app_bin}")
+
+  add_custom_target(mcub_conf DEPENDS ${mcuboot_build_dir}/mcuboot.defconfig)
+
+  ExternalProject_Add(
+    McuBoot
+    SOURCE_DIR ${mcuboot_esp_dir}
+    BINARY_DIR ${mcuboot_build_dir}
+    CONFIGURE_COMMAND ""
+    ${CMAKE_COMMAND} -G${CMAKE_GENERATOR}
+    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_CURRENT_LIST_DIR}/mcuboot-esp32.cmake
+    -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
+    -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
+    -DCMAKE_SYSTEM_NAME=${CMAKE_SYSTEM_NAME}
+    -DMCUBOOT_TARGET=esp32 -DMCUBOOT_CONFIG_FILE=${mcuboot_build_dir}/mcuboot.defconfig
+    -DIDF_PATH=${ESP_IDF_PATH} -S ${mcuboot_esp_dir}
+    BUILD_COMMAND
+    ${CMAKE_COMMAND} --build .
+    INSTALL_COMMAND ""
+    BUILD_ALWAYS 1
+    DEPENDS mcub_conf
+    )
+
+  set_property(GLOBAL APPEND PROPERTY extra_post_build_commands
+    COMMAND
+    ${PYTHON_EXECUTABLE} ${ESP_IDF_PATH}/components/esptool_py/esptool/esptool.py
```

Comment:
Where do I find `esptool.py` ?

It's not here: 
https://github.com/zephyrproject-rtos/zephyr/blob/9bea59a8c4b70ebd6da5fe2cd1566eb93ebbbdcf/boards/common/esp32.board.cmake#L13

```
/usr/bin/python3.8: can't open file '/projects/github/ncs/modules/hal/espressif/components/esptool_py/esptool/esptool.py': [Errno 2] No such file or directory
ninja: build stopped: subcommand failed.
```

Note, Zephyr doesn't clone submodules.
If this is required toll, please ensure it's part of the repositories handled be `west`.

Reply:
esptool is part of the submodules in `hal_espressif`.  
You have got to run `west espressif update` to retrieve that. It is part of the ESP32 guidelines.